js/tests/Test_jsoo.ml

```diff
@@ -17,88 +17,78 @@
 (* Prelude *)
 (*****************************************************************************)
 open Js_of_ocaml
-open Semgrep_js_shared
 
 (*****************************************************************************)
 (* Code *)
 (*****************************************************************************)
 
-(* These tests are skipped as they are currently incredibly slow *)
-(* Once LLVM merges a PR we will re-enable these *)
-let filtered = [ "Cpp"; "C++"; "Julia"; "Ruby" ]
+(* skipped_tests is a list of test names and and optional indicies to skip *)
+(* for example: *)
+(* ("foo", []) will skip all tests with "foo" in the name *)
+(* ("foo", [1; 3]) will skip test with "fool" in the name AND whose index are 1 or 3 *)
+let skipped_tests =
+  [
+    (* TODO: investigate C++ test issues *)
+    ("Cpp", []);
+    ("C++", []);
+    (* TODO: re-enable once we fix Julia build slowness *)
+    ("Julia", []);
+    (* TODO: re-enable once we fix Ruby build slowness *)
+    ("Ruby", []);
+    (* TODO: re-enable this when we fix the jsoo int overflow bug *)
+    ("Go", [ 24 ]);
+    (* TODO: investigate c_array_inits pattern parse error*)
+    ("C", [ 0 ]);
+  ]
 
 (* Filter to skip tests *)
 let test_filter ~name ~index =
-  ignore index;
-
-  if List.mem name filtered then `Skip
-  else `Run (* Cpp has a weird error, and @brandon is still working on it soo *)
+  if
+    List.filter
+      (fun (language, indexes) ->
+        Common.contains
+          (String.lowercase_ascii name)
+          (String.lowercase_ascii language)
+        && (indexes == [] || List.exists (fun n2 -> n2 == index) indexes))
```

Comment:
that was the location osemgrep was complaining (badly) about. In OCaml == is a physical equality, not structureal equality